migration/migration.go

```diff
@@ -110,18 +137,22 @@ func MigrateIfNeeded(targetDB db.DB, network utils.Network, log utils.SimpleLogg
 	return nil
 }
 
-func SchemaVersion(targetDB db.DB) (uint64, error) {
-	version := uint64(0)
+// SchemaMetadata retrieves metadata about a database schema from the given database.
+func SchemaMetadata(targetDB db.DB) (schemaMetadata, error) {
+	metadata := schemaMetadata{}
 	txn := targetDB.NewTransaction(false)
-	err := txn.Get(db.SchemaVersion.Key(), func(bytes []byte) error {
-		version = binary.BigEndian.Uint64(bytes)
+	err := txn.Get(db.SchemaVersion.Key(), func(b []byte) error {
+		err := metadata.fromBytes(b)
+		if err != nil {
+			metadata.Version = binary.BigEndian.Uint64(b)
```

Comment:
Can we assume that all unmarshal errors will exist because of our new way of representing the schema? I assume cbor can fail for other reasons.

Do we need a migration to update the schema representation? It may be easier to use a separate key in the database for `IntermediateState` than store all metadata in the same struct.